File: app/src/main/java/pl/orbitemobile/wspolnoty/activities/where/WhereContract.java
package pl.orbitemobile.wspolnoty.activities.where;

import android.view.MenuItem;

import pl.orbitemobile.wspolnoty.BasePresenter;
import pl.orbitemobile.wspolnoty.BaseView;

public class WhereContract {
    
    interface View extends BaseView<Presenter> {
        
    }
    
    interface Presenter extends BasePresenter {
        
        boolean onOptionsItemSelected(final MenuItem item);
        
        void onMapClick();
    }
}

File: app/src/main/java/pl/orbitemobile/wspolnoty/activities/where/WhereContract.kt
/*
 * Copyright (c) 2017. All Rights Reserved. Michal Jankowski orbitemobile.pl
 */

package pl.orbitemobile.wspolnoty.activities.where

import pl.orbitemobile.mvp.MvpPresenter
import pl.orbitemobile.mvp.MvpView

class WhereContract {

    abstract class View(layoutId: Int) : MvpView<Presenter>(layoutId)

    interface Presenter : MvpPresenter<View> {
        fun onMapClick()
    }
}
